[M]658.cpp

```diff
@@ -0,0 +1,45 @@
+// Problem: 658. Find K Closest Elements - Medium
+// Link: https://leetcode.com/problems/find-k-closest-elements/
+
+/* Nhận xét:
+- Anh ko hiểu bài này làm binary search kiểu gì luôn đấy. Chú thử gợi ý anh xem?
+- Anh submit 5 lần, lần nào cũng beat 100%, phải chăng bài này làm cách bình thường thì nhanh hơn binary search?
+*/
+
+class Solution {
+public:
+    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
+        int n = arr.size();
+        int l = 0, r = n - 1;
+        while (r - l > 0) {
+            int mid = l + (r - l) / 2;
+            if (arr[mid] >= x) {
+                r = mid;
+            } else {
+                l = mid + 1;
+            }
+        }
+        
+        int i = r, j = r - 1;
+        deque<int> res;
+        while (i < n || j >=0) {
+            if (i < n) {
```

Comment:
Cái đoạn if này hơi lằng nhằng. Là a thì a sẽ làm code nó clean thế này. 
while(i< n && j >= 0) {
   if (x - arr[j] <= arr[i] - x) {
     res.push_front(arr[j--]);
   } else {
     res.push_back(arr[i++]);
   }
}
//Đến đây thì i == n hoặc j == 0
while(i < n) res.push_back(arr[i++]); // vét hết các giá trị từ i đến cuối dãy nếu có thể
while(j > 0) res.push_front(arr[j--]); // vét hết các giá trị từ j đến đầu dãy nếu có thể.

Code nhìn sẽ khá clean.

Reply:
I see. Thank you for your suggestion :D
Đã fix, kiểm tra lại nhé :)

Reply:
ok roi day :))